packages/legend-query/src/stores/QueryBuilderPropertySearchPanelState.ts

```diff
@@ -0,0 +1,398 @@
+          if (
+            parentNode instanceof QueryBuilderExplorerTreePropertyNodeData &&
+            (parentNode.property.multiplicity.upperBound === undefined ||
+              parentNode.property.multiplicity.upperBound > 1)
+          ) {
+            return false;
+          }
+        }
+        return true;
+      }),
+    );
+  }
+
+  removeClassPropertyNodes(): void {
+    this.setClassMappedPropertyNodes(
+      this.searchedMappedPropertyNodes.filter((node) => {
+        if (node.type instanceof Class) {
+          return false;
+        }
+        return true;
+      }),
+    );
+  }
+
+  removeStringPropertyNodes(): void {
+    this.setStringMappedPropertyNodes(
+      this.searchedMappedPropertyNodes.filter((node) => {
+        if (
+          node.type instanceof PrimitiveType &&
+          node.type.name === PRIMITIVE_TYPE.STRING
+        ) {
+          return false;
+        }
+        return true;
+      }),
+    );
+  }
+
+  removeNumberPropertyNodes(): void {
+    this.setNumberMappedPropertyNodes(
+      this.searchedMappedPropertyNodes.filter((node) => {
+        if (
+          node.type instanceof PrimitiveType &&
+          (node.type.name === PRIMITIVE_TYPE.DECIMAL ||
+            node.type.name === PRIMITIVE_TYPE.NUMBER ||
+            node.type.name === PRIMITIVE_TYPE.INTEGER ||
+            node.type.name === PRIMITIVE_TYPE.FLOAT)
+        ) {
+          return false;
+        }
+        return true;
+      }),
+    );
+  }
+
+  removeDatePropertyNodes(): void {
+    this.setDateMappedPropertyNodes(
+      this.searchedMappedPropertyNodes.filter((node) => {
+        if (
+          node.type instanceof PrimitiveType &&
+          (node.type.name === PRIMITIVE_TYPE.DATE ||
+            node.type.name === PRIMITIVE_TYPE.DATETIME ||
+            node.type.name === PRIMITIVE_TYPE.STRICTDATE ||
+            node.type.name === PRIMITIVE_TYPE.STRICTTIME ||
+            node.type.name === PRIMITIVE_TYPE.LATESTDATE)
+        ) {
+          return false;
+        }
+        return true;
+      }),
+    );
+  }
+
+  removeBooleanPropertyNodes(): void {
+    this.setBooleanMappedPropertyNodes(
+      this.searchedMappedPropertyNodes.filter((node) => {
+        if (
+          node.type instanceof PrimitiveType &&
+          node.type.name === PRIMITIVE_TYPE.BOOLEAN
+        ) {
+          return false;
+        }
+        return true;
+      }),
+    );
+  }
+
+  updateOneManyRowsIncludedMappedProperties(): void {
+    this.isOneManyRowsIncluded
+      ? this.setOneManyMappedPropertyNodes(this.searchedMappedPropertyNodes)
+      : this.removeOneManyPropertyNodes();
+  }
+
+  updateClassIncludedMappedProperties(): void {
+    this.isClassIncluded
+      ? this.setClassMappedPropertyNodes(this.searchedMappedPropertyNodes)
+      : this.removeClassPropertyNodes();
+  }
+
+  updateStringIncludedMappedProperties(): void {
+    this.isStringIncluded
+      ? this.setStringMappedPropertyNodes(this.searchedMappedPropertyNodes)
```

Comment:
I find this code a bit confusing. You are setting all the `searchedMappedPropertyNodes` as the stringMappedPropertyNodes. You do this for types of nodes. This causes the method `filteredPropertyNodes` to be a bit confusing to read.
Couldn't these all be `computed` from the seachMappedPropertyNodes? 
You can then just use the flag of whether the user has checked off that type to decide whether to show it or not.